src/tools/w3dviewer/mainfrm.cpp

```diff
@@ -0,0 +1,1607 @@
+{
+    CGraphicView *view = reinterpret_cast<CGraphicView *>(m_splitter.GetPane(0, 1));
+
+    if (view != nullptr) {
+        pCmdUI->SetCheck(view->m_cameraRotateConstraints == CGraphicView::ROTATE_Z_ONLY);
+    }
+}
+
+void CMainFrame::OnUpdateRotateX(CCmdUI *pCmdUI)
+{
+    CGraphicView *view = reinterpret_cast<CGraphicView *>(m_splitter.GetPane(0, 1));
+
+    if (view != nullptr) {
+        pCmdUI->SetCheck(view->m_objectRotationFlags & CGraphicView::ROTATE_PLUSX);
+    }
+}
+
+void CMainFrame::OnUpdateRotateY(CCmdUI *pCmdUI)
+{
+    CGraphicView *view = reinterpret_cast<CGraphicView *>(m_splitter.GetPane(0, 1));
+
+    if (view != nullptr) {
+        pCmdUI->SetCheck(view->m_objectRotationFlags & CGraphicView::ROTATE_PLUSY);
+    }
+}
+
+void CMainFrame::OnUpdateRotateZ(CCmdUI *pCmdUI)
+{
+    CGraphicView *view = reinterpret_cast<CGraphicView *>(m_splitter.GetPane(0, 1));
+
+    if (view != nullptr) {
+        pCmdUI->SetCheck(view->m_objectRotationFlags & CGraphicView::ROTATE_PLUSZ);
+    }
+}
+
+void CMainFrame::OnUpdateEditEmitter(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateExportEmitter(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateAnimateCamera(CCmdUI *pCmdUI)
+{
+    CW3DViewDoc *doc = reinterpret_cast<CW3DViewDoc *>(GetActiveDocument());
+    pCmdUI->SetCheck(doc->m_animateCamera);
+}
+
+void CMainFrame::OnUpdateExportLod(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateExportAggregate(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateResetOnDisplay(CCmdUI *pCmdUI)
+{
+    CW3DViewDoc *doc = reinterpret_cast<CW3DViewDoc *>(GetActiveDocument());
+    pCmdUI->SetCheck(doc->m_resetCamera);
+}
+
+void CMainFrame::OnUpdateIncludeNull(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdatePrevLod(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateNextLod(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateAutoSwitching(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateMovie(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateAnimAdvanced(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateCopyDeps(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateExposePrecalc(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateEditPrimitive(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateExportPrimitive(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateMultiPassLighting(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateMultiTextureLighting(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateVertexLighting(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateAddObject(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateImportFacial(CCmdUI *pCmdUI)
+{
+    // do later
+}
+
+void CMainFrame::OnUpdateRestrictAnims(CCmdUI *pCmdUI)
+{
+    bool restrict = true;
```

Comment:
`restrict` is c++ keyword. Maybe use different name.

Reply:
fixed.